tasks/certificate.yml

```diff
@@ -0,0 +1,66 @@
+---
+- name: Generate certificate using PostgreSQL role
+  block:
+    - name: Generate certificates
+      include_role:
+        name: fedora.linux_system_roles.certificate
+      vars:
+        certificate_requests: "{{ postgresql_certificates }}"
+
+    - name: Install certificate from the default path
+      file:
+        src: >-
+          /etc/pki/tls/certs/{{ ( postgresql_certificates |
+          first )['name'] }}.crt
+        dest: /var/lib/pgsql/data/server.crt
+        state: link
+        owner: postgres
+      when: ( postgresql_certificates | first )['name'] is not abs
+
+    - name: Install certificate from the default path
+      file:
+        src: >-
+          /etc/pki/tls/private/{{ ( postgresql_certificates |
+          first )['name'] }}.key
+        dest: /var/lib/pgsql/data/server.key
+        state: link
+        owner: postgres
+      when: ( postgresql_certificates | first )['name'] is not abs
+
+    - name: Install certificate from custom path
+      file:
+        src: "{{ ( postgresql_certificates | first )['name'] }}.crt"
+        dest: /var/lib/pgsql/data/server.crt
+        state: link
+        owner: postgres
+      when: ( postgresql_certificates | first )['name'] is abs
+
+    - name: Install certificate from custom path
+      file:
+        src: "{{ ( postgresql_certificates | first )['name'] }}.key"
+        dest: /var/lib/pgsql/data/server.key
+        state: link
+        owner: postgres
+      when: ( postgresql_certificates | first )['name'] is abs
+
+  when: postgresql_certificates | length > 0
+
+- name: Install user provided TLS certificates for postgresql
+  block:
+    - name: Install certificate file
+      file:
+        src: "{{ postgresql_cert_name }}.crt"
+        dest: /var/lib/pgsql/data/server.crt
+        state: link
+        owner: postgres
+
+    - name: Install postgresql server private key
+      file:
+        src: "{{ postgresql_cert_name }}.key"
+        dest: /var/lib/pgsql/data/server.key
+        state: link
+        owner: postgres
+  when:
```

Comment:
please put the `when` condition for the `block` between `name` and `block` - it improves readability